### PR TITLE
Docs: Plugin E2E - Add annotations article

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

Another article in the data source plugin guide. This time on how to test annotation queries. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+id: annotation-queries
+title: Annotation queries
+description: Test a custom annotation editor and the execution of annotation queries
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:4]:
```suggestion
description: Test a custom annotation editor and the execution of annotation queries.
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  from dataset
+  where $__timeFilter(time) and humidity > 95`);
+  await expect(annotationEditPage.runQuery()).toBeOK();
+  await expect(annotationEditPage).toHaveAlert('success');
+});
+```
+
+#### Testing error scenarios
+
+If an error occurs in the plugin or if the upstream API returns an error, you may want to capture that and return a meaningful error message to the user.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:49]:
```suggestion
If an error occurs in the plugin or if the upstream API returns an error, we recommend that you capture that error and return a meaningful error message to the user.
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+Annotations can be used to mark points on a visualization with events such as `AB test started` and `Marketing campaign started`. If a data source plugin supports annotations, it means that the data source can be used to query for annotation events. Optionally, data sources plugins can implement a custom annotation editor that can be used to craft the annotation query. In many cases, the execution of annotation queries requires different handling than normal data queries, and in those cases you may want to write end-to-end tests that verifies that data source annotation support work as expected.
+
+### Testing the annotation editor
+
+If your data source plugin implements a custom annotation editor, you can write tests that verifies that the editor works as expected. If no custom editor is implemented, the built-in in editor will be used and then you don't need to write tests for that.
+
+### Testing the entire annotation query execution flow
+
+In the next example, we perform an integration test where we test a plugin's entire annotation query data flow.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:26]:
```suggestion
In the next example, we perform an integration test where we test a plugin's entire annotation query data flow:
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+sidebar_position: 50
+---
+
+## Introduction
+
+Annotations can be used to mark points on a visualization with events such as `AB test started` and `Marketing campaign started`. If a data source plugin supports annotations, it means that the data source can be used to query for annotation events. Optionally, data sources plugins can implement a custom annotation editor that can be used to craft the annotation query. In many cases, the execution of annotation queries requires different handling than normal data queries, and in those cases you may want to write end-to-end tests that verifies that data source annotation support work as expected.
+
+### Testing the annotation editor
+
+If your data source plugin implements a custom annotation editor, you can write tests that verifies that the editor works as expected. If no custom editor is implemented, the built-in in editor will be used and then you don't need to write tests for that.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:22]:
```suggestion
If your data source plugin implements a custom annotation editor, you can write tests that verify that the editor works as expected. If you haven't implemented a custom editor, then the plugin will use the built-in in editor. In that case, you don't need to write tests.
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  - testing
+  - e2e
+  - data-source
+  - annotation queries
+sidebar_position: 50
+---
+
+## Introduction
+
+Annotations can be used to mark points on a visualization with events such as `AB test started` and `Marketing campaign started`. If a data source plugin supports annotations, it means that the data source can be used to query for annotation events. Optionally, data sources plugins can implement a custom annotation editor that can be used to craft the annotation query. In many cases, the execution of annotation queries requires different handling than normal data queries, and in those cases you may want to write end-to-end tests that verifies that data source annotation support work as expected.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:18]: Re: " In many cases, the execution of annotation queries requires different handling than normal data queries..." - Do we need to say anything about those cases which don't require different handling/testing?

**Review by josmperez — APPROVED** [review]
I've made several editorial suggestions, but I don't see any blocking issues.

**Review by mckn — APPROVED** [review]
LGTM!

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:18]: basically they don't need to write tests, so think this is fine ^.

**Pull request merged**